test/plugin/test_out_http.rb

```diff
@@ -0,0 +1,351 @@
+
+  def create_driver(conf)
+    Fluent::Test::Driver::Output.new(Fluent::Plugin::HTTPOutput).configure(conf)
+  end
+
+  def config
+    %[
+      endpoint #{base_endpoint}/test
+    ]
+  end
+
+  def test_events
+    [
+      {"message" => "hello", "num" => 10, "bool" => true},
+      {"message" => "hello", "num" => 11, "bool" => false}
+    ]
+  end
+
+  def test_configure
+    d = create_driver(config)
+    assert_equal "http://127.0.0.1:#{server_port}/test", d.instance.endpoint
+    assert_equal :post, d.instance.http_method
+    assert_equal 'application/x-ndjson', d.instance.content_type
+    assert_equal [503], d.instance.retryable_response_codes
+    assert_true d.instance.error_response_as_unrecoverable
+    assert_nil d.instance.proxy
+    assert_nil d.instance.headers
+  end
+
+  data('json' => ['json', 'application/x-ndjson'],
+       'ltsv' => ['ltsv', 'text/tab-separated-values'],
+       'msgpack' => ['msgpack', 'application/x-msgpack'],
+       'single_value' => ['single_value', 'text/plain'])
+  def test_configure_content_type(types)
+    format_type, content_type = types
+    d = create_driver(config + %[
+      <format>
+        @type #{format_type}
+      </format>
+    ])
+    assert_equal content_type, d.instance.content_type
+  end
+
+  data('PUT' => 'put', 'POST' => 'post')
+  def test_write_with_method(method)
+    d = create_driver(config + "http_method #{method}")
+    d.run(default_tag: 'test.http') do
+      test_events.each { |event|
+        d.feed(event)
+      }
+    end
+
+    result = @@result
+    assert_equal method.upcase, result.method
+    assert_equal 'application/x-ndjson', result.content_type
+    assert_equal test_events, result.data
+    assert_not_empty result.headers
+  end
+
+  def test_write_with_single_value_format
+    d = create_driver(config + %[
+      <format>
+        @type single_value
+      </format>
+    ])
+    d.run(default_tag: 'test.http') do
+      test_events.each { |event|
+        d.feed(event)
+      }
+    end
+
+    result = @@result
+    assert_equal 'text/plain', result.content_type
+    assert_equal (test_events.map { |e| e['message'] }), result.data
+    assert_not_empty result.headers
+  end
+
+  def test_write_with_headers
+    d = create_driver(config + 'headers {"test_header":"fluentd!"}')
+    d.run(default_tag: 'test.http') do
+      test_events.each { |event|
+        d.feed(event)
+      }
+    end
+
+    result = @@result
+    assert_true result.headers.has_key?('test_header')
+    assert_equal "fluentd!", result.headers['test_header']
+  end
+
+  def test_write_with_retryable_response
+    d = create_driver("endpoint #{base_endpoint}/503")
+    assert_raise(Fluent::Plugin::HTTPOutput::RetryableResponse) do
+      d.run(default_tag: 'test.http', shutdown: false) do
+        test_events.each { |event|
+          d.feed(event)
+        }
+      end
+    end
+    d.instance_shutdown
+  end
+
+  # basic auth test includes "error_response_as_unrecoverable true" case
```

Comment:
unnecessary comment?

Reply:
I think this is needed because I don't add explicit standalone test for `true` version

Reply:
> I think this is needed because I don't add explicit standalone test for `true` version

Yes. but 

> basic auth test includes "error_response_as_unrecoverable true" case

what does this comment mean?
This test is not for basic auth, right? and basic auth test looks not having `error_response_as_unrecoverable true` case. why do we need this comment?